ViewModel/NavigationViewModel.swift

```diff
@@ -20,6 +20,8 @@ class NavigationViewModel: ObservableObject {
         let tab = Tab(context: context)
         tab.created = Date()
         tab.lastOpened = Date()
```

Comment:
We were using the same set of operations after creating a tab:
- obtaining an DB id for it
- saving it
so it makes sense to bring those under the same function.